openmetadata-ui/src/main/resources/ui/src/components/Ingestion/AddIngestionButton.component.tsx

```diff
@@ -0,0 +1,170 @@
+import { MetadataServiceType } from '../../generated/api/services/createMetadataService';
+import { PipelineType } from '../../generated/entity/services/ingestionPipelines/ingestionPipeline';
+import { getAddIngestionPath } from '../../utils/RouterUtils';
+import { dropdownIcon as DropdownIcon } from '../../utils/svgconstant';
+import DropDownList from '../dropdown/DropDownList';
+import { AddIngestionButtonProps } from './ingestion.interface';
+
+function AddIngestionButton({
+  serviceDetails,
+  pipelineType,
+  serviceCategory,
+  serviceName,
+  ingestionData,
+  ingestionList,
+  permissions,
+}: AddIngestionButtonProps) {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const [showActions, setShowActions] = useState(false);
+
+  const isOpenMetadataService = useMemo(
+    () =>
+      serviceDetails?.connection?.config?.type ===
+      MetadataServiceType.OpenMetadata,
+    [serviceDetails]
+  );
+
+  const supportedPipelineTypes = useMemo(
+    (): PipelineType[] => getSupportedPipelineTypes(serviceDetails),
+    [serviceDetails]
+  );
+
+  const handleAddIngestionClick = (type?: PipelineType) => {
+    setShowActions(false);
+    if (type) {
+      history.push(getAddIngestionPath(serviceCategory, serviceName, type));
+    }
+  };
+
+  const isDataSightIngestionExists = useMemo(
+    () =>
+      ingestionData.some(
+        (ingestion) => ingestion.pipelineType === PipelineType.DataInsight
+      ),
+    [ingestionData]
+  );
+
+  const types = useMemo(
+    (): PipelineType[] =>
+      getIngestionTypes(
+        supportedPipelineTypes,
+        isOpenMetadataService,
+        ingestionList,
+        pipelineType
+      ),
+    [pipelineType, supportedPipelineTypes, isOpenMetadataService, ingestionList]
+  );
+
+  // Check if service has atleast one metadata pipeline available or not
+  const hasMetadata = useMemo(
+    () =>
+      ingestionList.find(
+        (ingestion) => ingestion.pipelineType === PipelineType.Metadata
+      ),
+    [ingestionList]
+  );
+  if (types.length === 0) {
+    return null;
+  }
+
+  // if service has metadata then show all available option
+  if (hasMetadata) {
+    return (
+      <>
+        <Button
+          className={classNames('h-8 rounded-4 m-b-xs d-flex items-center')}
+          data-testid="add-new-ingestion-button"
+          disabled={!permissions.Create}
+          size="small"
+          type="primary"
+          onClick={() => setShowActions((pre) => !pre)}>
+          {t('label.add-entity', { entity: t('label.ingestion-lowercase') })}
+          {showActions ? (
+            <DropdownIcon
+              style={{
+                transform: 'rotate(180deg)',
+                verticalAlign: 'middle',
+                color: '#fff',
+              }}
+            />
+          ) : (
+            <DropdownIcon
+              style={{
+                color: '#fff',
+                verticalAlign: 'middle',
+              }}
+            />
+          )}
```

Comment:
Use button's icon prop instead

Reply:
Didn't use the icon prop of the button as it doesn't allow configuring the position of the icon to display.